core/src/main/java/org/opensearch/sql/data/model/ExprValueUtils.java

```diff
@@ -115,6 +137,16 @@ public static ExprValue fromObjectValue(Object o) {
       return stringValue((String) o);
     } else if (o instanceof Float) {
       return floatValue((Float) o);
+    } else if (o instanceof LocalDate) {
```

Comment:
omg, can we turn this into a switch statement on an enum please? 

https://stackoverflow.com/questions/5579309/is-it-possible-to-use-the-instanceof-operator-in-a-switch-statement

Reply:
Unfortunately, this doesn't work with interfaces or any inherited classes. Instead of `instanceof Map` I need to list all possible `Map` implementations.